### ISSUE TITLE
naming and metadata keys in derivatives

### ISSUE THREAD

**Issue by satra** [bids-standard/bids-specification#309]
cc/ @bids-standard/derivatives

this is an offshoot of #301 

at the bids derivatives meeting one thought was the following:

```
Consider storing all metadata in the json file, bring them to the filename as 
needed only for filesystem disambiguation, but not processing. This means 
that filename globbing is not a recommended pattern outside of the base 
filename. Indexing and reading for processing will require reading the full 
metadata dictionary. 
```

so while we may be able to solve the current state with some tweaks of keywords, there is a more fundamental problem of categorization/parameterization.  what if i run a pipeline that compares spm/fsl/afni models like the work from @nicholst? or one that compares os environments? or essentially any parametric/algorithmic/environment mixture. 

is there some expected consensus that:
1. these are out of scope and cannot be stored as bids derivatives? 
2. any parameterization should be stored as separate derivative datasets?

if not, could the above principle work?

The principle could dissociate the creation of keys (knowledge) from the current use of keys as glob patterns and address disambiguation in a systematic manner.

**Comment by sappelhoff** [comment]
@effigies will this be closed upon merging #265? If yes, could you please update the PR description of #265 to include this as an auto-close syntax?

**Comment by effigies** [comment]
No, this hasn't been directly addressed in #265. I don't think it was necessary to invoke, and I don't think moving toward it in the future will break compatibility with the principles established in #265.
